Spock-core/src/Web/Spock/Core.hs

```diff
@@ -89,16 +91,33 @@ spockT :: (MonadIO m)
        => (forall a. m a -> IO a)
        -> SpockT m ()
        -> IO Wai.Middleware
-spockT = spockLimT Nothing
+spockT = spockConfigT defaultSpockConfig
 
 -- | Like @spockT@, but first argument is request size limit in bytes. Set to 'Nothing' to disable.
+-- | This function is deprecated. Consider using @spockConfigT@ instead.
 spockLimT :: forall m .MonadIO m
        => Maybe Word64
        -> (forall a. m a -> IO a)
        -> SpockT m ()
        -> IO Wai.Middleware
-spockLimT mSizeLimit liftFun app =
-    W.buildMiddleware mSizeLimit liftFun (baseAppHook app)
+spockLimT mSizeLimit  =
```

Comment:
We could use `{-# DEPRECATED spockLimT "Consider using spockConfigT" #-}` here to mark this as deprecated